documentation/library_guide/parallel_api/iterators.rst

```diff
@@ -69,10 +69,18 @@ header.  All iterators are implemented in the ``oneapi::dpl`` namespace.
 * ``transform_iterator``: an iterator defined over another iterator whose dereferenced value is the result
   of a function applied to the corresponding element of the original iterator. Both the type of the original
   iterator and the unary function applied during dereference operations are required template parameters of
-  the ``transform_iterator`` class. The ``transform_iterator`` class provides three constructors:
+  the ``transform_iterator`` class. 
+
+  The unary functor provided to a transform iterator should have a const call operator which is callable with an
+  argument of the reference type of the original iterator. The functor's call operator should not have any side
```

Comment:
> an argument of the reference type of the original iterator...

That is something I don't understand

Reply:
Basically, I mean to say that we want `__my_unary_func_(*__my_it_)` to be a valid call.  
`*__my_it_` returns a reference to an element of the original iterator, we want to be able to use that to call the callable (directly or via some conversion sequence).

Perhaps the sentence should instead read: 
"The unary functor provided to a ``transform_iterator`` should have a ``const``-qualified call operator that can be called using the reference type of the original iterator as argument."

Reply:
I adjusted it as mentioned above, let me know if it still requires clarification.